File: ipman.shared/Entity/SiteAccount.cs
using ipman.shared.Entity.Join;
using ipman.shared.Entity.Lookups;
using System;
using System.Collections.Generic;
using System.Text;

namespace ipman.shared.Entity
{
    public class SiteAccount: EntityBase
    {
        public string SiteAccountName { get; set; }
        public string SiteAccountImagePath { get; set; }
        public string SiteAccountThemeColorPrimary { get; set; }
        public string SiteAccountThemeColorSecondary { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreatedUTC { get; set; }
        public DateTime LastUpdatedUTC { get; set; }
        public SiteAccountType SiteAccountType { get; set; }
        public virtual ICollection<SiteAccountUserAccount> SiteAccountUserAccounts { get; set; }
        public virtual ICollection<Department> Departments { get; set; }
        public virtual ICollection<Post> Posts { get; set; }
        public virtual ICollection<Tag> Tags { get; set; }
    }
}

File: ipman/ClientApp/entity/SiteAccount.ts
import { EntityBase } from "./EntityBase";
import SiteAccountUserAccount from "./join/SiteAccountUserAccount";
import Department from "./Department";
import Post from "./Post";
import Tag from "./Tag";

export default interface SiteAccount extends EntityBase
{
    SiteAccountName: string;
    SiteAccountImagePath: string; 
    SiteAccountThemeColorPrimary: string;
    SiteAccountThemeColorSecondary: string;
    SiteAccountType: any;
    IsActive: boolean;
    CreatedUTC: Date | string;
    LastUpdatedUTC: Date | string;
    SiteAccountUserAccounts: SiteAccountUserAccount[];
    Departments: Department[];
    Posts: Post[];
    Tags: Tag[];
}
